File: LA/Baseline-CQCC-GMM/matlab/CQCC/CQT_toolbox_2013/hp.m
function b = hp(fs)

% Equiripple Highpass filter designed using the FIRPM function.
% All frequency values are in Hz.

Fstop = 0.125;           % Stopband Frequency
Fpass = 0.25;            % Passband Frequency
Dstop = 0.001;           % Stopband Attenuation
Dpass = 0.057501127785;  % Passband Ripple
dens  = 20;              % Density Factor

% Calculate the order from the parameters using FIRPMORD.
[N, Fo, Ao, W] = firpmord([Fstop, Fpass]/(fs/2), [0 1], [Dstop, Dpass]);

% Calculate the coefficients using the FIRPM function.
b  = firpm(N, Fo, Ao, W, {dens});

end

File: LA/Baseline-CQCC-GMM/python/CQCC/CQT_toolbox_2013/hp.py
import numpy as np

'''
1. firpmord
2. firpm
'''

def hp(fs):

    Fstop = 0.125
    Fpass = 0.25
    Dstop = 0.001
    Dpass = 0.057501127785
    dens  = 20

    # firpmord
    N, Fo, Ao, W = firpmord([Fstop, Fpass]/(fs/2), [0 1], [Dstop, Dpass])

    # firpm
    b  = firpm(N, Fo, Ao, W, {dens})

    return b
